src/rust/generators/sysmon-generator/src/serialization.rs

```diff
@@ -30,12 +31,19 @@ impl CheckedError for SysmonDecoderError {
     }
 }
 
+impl From<prost::DecodeError> for SysmonDecoderError {
+    fn from(err: prost::DecodeError) -> Self {
+        SysmonDecoderError::DeserializeError(err.to_string())
+    }
+}
+
```

Comment:
The generator shouldn't care about grapl service proto, it should only need to work with log data, but even still, this should be its own variant of `SysmonDecoderError` that uses thiserror's `#[from] prost::DecodeError`, instead of packing it into the `DeserializeError` variant.

Reply:
Removed